scalar/bool.go

```diff
@@ -67,6 +67,10 @@ func (s *Bool) Set(val any) error {
 	case *bool:
 		return s.Set(*value)
 	case *string:
+		if value == nil {
```

Comment:
Don't we need this check in the `*bool` case too?

Reply:
+100